catalog/core/catalog-core-standardframework/src/main/java/ddf/catalog/impl/operations/QueryOperations.java

```diff
@@ -204,6 +212,26 @@ QueryResponse doQuery(QueryRequest queryRequest, FederationStrategy strategy)
 
         LOGGER.debug("Calling strategy.federate()");
 
+        Query originalQuery = queryRequest.getQuery();
+
+        long timeout = originalQuery.getTimeoutMillis();
+
+        if (timeout == 0) {
+            timeout = queryTimeoutMillis;
```

Comment:
If that's the case I must've misunderstood the scope. I thought the purpose was to enforce timeouts altogether, not just enforce any timeouts already included in the query. I'll remove this logic then